draft-irtf-cfrg-opaque.md

```diff
@@ -734,6 +896,7 @@ data
 struct {
     SerializedElement data;
     opaque server_public_key[Npk];
+    opaque client_public_key[Npk];
```

Comment:
Why is this needed? In internal mode, clients can compute it from the seed. In external mode, clients can also presumably compute it from the private key. Is this for settings where clients can't compute the public key from a private key?

Reply:
Indeed. I don't know if we can presume that clients can recompute their public key from the private key alone.
As much as I dislike this, simply sending it back to the client covers all cases and simplifies the implementation.

But if you think we should assume public key regeneration possible from the private one I'm totally on board.

Reply:
Also, deriving the public key from the private key pushes the `auth_tag` verification to be done after the secret is recovered, i.e. we have the secret open and in memory, before we can authenticate the envelope. But maybe I'm just being overly paranoid.

Reply:
Hmm... how does this simplify the implementation? Does the client need to check that the public key it receives is correct for the given private key?

Reply:
It simplifies the implementation in case you're using something else than POG and if you need extra data to generate your public key (e.g. for RSA). But I forgot we pinned this to 3DH only. I'll amend.

Reply:
Fixed. See https://github.com/cfrg/draft-irtf-cfrg-opaque/pull/154/commits/ad77e8accc08c246dca858aeb9fbf503566ec506